wp-content/themes/engage/src/Models/Homepage.php

```diff
@@ -36,35 +38,104 @@ public function getRecent(){
+		}
+
+        // sort the posts by their time
+        $this->sortByDate($this->recent);
+        $this->sortByDate($this->moreRecent);
+    }
 
-            $verticalName = $verticals[$i]->slug;
+    //get the most recent featured research
+    public function getRecentFeaturedResearch($verticalName){
+        $featuredPosts = $this->queryPosts(true, $verticalName);
+        $recentFeaturedPosts = array();
+        //only show one featured research per vertical;
+		foreach($featuredPosts as $featurePost){
+			array_push($recentFeaturedPosts, $featurePost);
+			break;
+		}
+        return $recentFeaturedPosts;
+    }
 
-            // Get the most recent post for that specific vertical
-            $queryResults = $this->Query->getRecentPosts([
-                'postType' => 'research',
-                'vertical' => $verticalName,
-                'postsPerPage' => 3,
-            ]);
+    //get the more research posts
+	public function getMoreRecentResearch($featuredSliderPosts, $verticalName){
+        // how many more_research_posts should be display on the home page for each vertical
+        $numFeaturedPerVertical = [
+            "journalism" => 4,
+            "media-ethics" => 2,
+            "science-communication" => 2
+        ];
+        $allRecentResearch = $this->queryPosts(false, $verticalName);
+        $moreRecentResearch = array();
+        for($i = 0; $i<count($allRecentResearch) && $numFeaturedPerVertical[$verticalName] > 0; $i++){
+            $curr = $allRecentResearch[$i];
+            // make sure not already displaying the post, and that we haven't used up the allotted num of posts
+            if($this->notInSlider($curr, $featuredSliderPosts)){
+                $numFeaturedPerVertical[$verticalName]--;
+                array_push($moreRecentResearch, $curr);
+            }
+        }
+        return $moreRecentResearch;
+	}
 
-            $tempArray = array($queryResults[0]);
-            $tempArrayMoreRecent = array($queryResults[1], $queryResults[2]);
+    // check if a post with the same id as curr is not on the slider
+    public function notInSlider($curr, $featuredSliderPosts){
+		foreach($featuredSliderPosts as $featurePost){
+			// posts with same id are the same post
+            if($curr->id == $featurePost->id){
+                return false;
+            }
+		}
+        return true;
+    }
 
-            // Merge the new post with the existing ones
-            // (Maybe there's a more efficient way to do this?)
-            $this->recent = array_merge($tempArray, $this->recent);
-            $this->moreRecent = array_merge($tempArrayMoreRecent, $this->moreRecent);
+    // query the posts with the given arguments
+    public function queryPosts($is_featured, $verticalName){
+        $args = [
+            'postType' => 'research',
+            'vertical' => $verticalName,
+            'postsPerPage' => '100',
```

Comment:
I remember we discussed changing this so that we were only requesting the number of posts we needed, rather than 100 of them. Can you work on that change before we merge this? If you need help with it, let me know.

Reply:
I definitely pushed up a version that used the meta query, I think it must have gotten reverted or something without noticing? I'll get it back up

Reply:
The meta query is still underneath of this code, but the postsPerPage wasn't modified to match whatever it is supposed to be.